### ISSUE TITLE
💅 lint/suspicious/noExtraNonNullAssertion triggers for `(a!)`

### ISSUE THREAD

**Issue by pfgithub** [biomejs/biome#3352]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.17.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.1.16"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  false
  Recommended:                  false
  All:                          false
  Enabled rules:
  suspicious/noExtraNonNullAssertion

Workspace:
  Open Documents:               0
```

### Rule name

lint/suspicious/noExtraNonNullAssertion

### Playground link

https://biomejs.dev/playground/?code=ZgB1AG4AYwB0AGkAbwBuACAAZABlAG0AbwAoAHYAYQBsAHUAZQA6ACAAcwB0AHIAaQBuAGcAIAB8ACAAbgB1AGwAbAApADoAIABzAHQAcgBpAG4AZwAgAHsACgAgACAAcgBlAHQAdQByAG4AIAAoAHYAYQBsAHUAZQAhACkAOwAKAH0A

### Expected result

Forbidden extra non-null assertion should not trigger, and definitely shouldn't be safely auto-fixable.

As it is, forbidden extra non-null assertion triggers and auto-fixes to remove the non-null assertion, despite it clearly being required

```
  ✖ Forbidden extra non-null assertion.
  
    1 │ function demo(value: string | null): string {
  > 2 │   return (value!);
      │           ^^^^^^
    3 │ }
    4 │ 
  
  ℹ Safe fix: Remove extra non-null assertion.
  
    2 │   return (value!);
      │                -  
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
